src/Fhir.Anonymizer.Core/AnonymizerEngine.cs

```diff
@@ -56,147 +67,46 @@ public static AnonymizerEngine CreateWithFileContext(string configFilePath, stri
-            }
+            ValidateInput(settings, resource);
+            var resourceNode = ElementNode.FromElement(resource.ToTypedElement());
+            var anonymizedResource = resourceNode.Anonymize(_rules, _processors).ToPoco<Resource>();
+            ValidateOutput(settings, anonymizedResource);
 
-            if (settings != null && settings.ValidateInput)
-            {
-                _validator.ValidateInput(resource);
-            }
+            return anonymizedResource;
+        }
 
-            var processResult = AnonymizeResourceNode(root);
-            root.RemoveNullChildren();
-            var anonymizedResource = root.ToPoco<Resource>();
-            anonymizedResource.TryAddSecurityLabels(processResult);
+        public string AnonymizeJson(string json, AnonymizerSettings settings = null)
+        {
+            EnsureArg.IsNotNullOrEmpty(json, nameof(json));
+
+            var resource = _parser.Parse<Resource>(json);
+            Resource anonymizedResource = AnonymizeResource(resource, settings);
 
-            if (settings != null && settings.ValidateOutput)
-            {
-                _validator.ValidateOutput(anonymizedResource);
-            }
-            
             FhirJsonSerializationSettings serializationSettings = new FhirJsonSerializationSettings
             {
                 Pretty = settings != null && settings.IsPrettyOutput
             };
             return anonymizedResource.ToJson(serializationSettings);
         }
 
-        public ProcessResult AnonymizeResourceNode(ElementNode root)
-        {
-            EnsureArg.IsNotNull(root, nameof(root));
-
-            var processResult = new ProcessResult();
-            if (root.IsBundleNode())
-            {
-                var entryResources = root.GetEntryResourceChildren();
-                var entryResult = AnonymizeBundleEntryResourceNodes(entryResources);
-                processResult.UpdateResult(entryResult);
-            }
-
-            if (root.HasContainedNode())
-            {
-                var containedResources = root.GetContainedChildren();
-                var containedResult = AnonymizeContainedResourceNodes(containedResources);
-                processResult.UpdateResult(containedResult);
-            }
-
-            var resourceContext = ResourceAnonymizerContext.Create(root, _configurationManger);
-            var resourceId = root.GetNodeId();
-            foreach (var rule in resourceContext.RuleList)
-            {
-                var matchedNodes = root.Select(rule.Path).Cast<ElementNode>();
-
-                _logger.LogDebug(rule.Type == AnonymizerRuleType.PathRule ?
-                    $"Path {rule.Source} matches {matchedNodes.Count()} nodes in resource ID {resourceId}." :
-                    $"Type {rule.Source} matches {matchedNodes.Count()} nodes with path {rule.Path} in resource ID {resourceId}.");
```

Comment:
Where do we print these verbose log in the new config?

Reply:
Thanks, added verbose log